### PR TITLE
Level manager (1 / 3)

### PR DESCRIPTION
**Pull request by skumbananaaa**
Adds a definition for LevelManager, the levels.json file, and how level loading works.
Also adds SHA256 hashing.

### CODE REVIEW

**Review by Vonsen — APPROVED** [review]
Looks good

File: LambdaEngine/Source/Resources/ResourceLoader.cpp [hunk 1]
```diff
@@ -1410,4 +1310,104 @@ namespace LambdaEngine
...
+
+		if (!program.link(messages))
+		{
+			LOG_ERROR("[ResourceLoader]: GLSL Linking failed for: \"%s\"\n%s\n%s", filepath.c_str(), shader.getInfoLog(), shader.getInfoDebugLog());
+			return false;
+		}
+
+		glslang::TIntermediate* pIntermediate = program.getIntermediate(shaderType);
+
+		String sourcesss = pIntermediate->getSourceText();
```
Comment (norda97) [inline LambdaEngine/Source/Resources/ResourceLoader.cpp]: Unused variable

Comment (norda97) [inline LambdaEngine/Source/Resources/ResourceLoader.cpp]:
```suggestion
		
```

**Review by norda97 — APPROVED** [review]
Looks good, only one unused variable

File: LambdaEngine/Source/Resources/ResourceLoader.cpp (hunk 1 above)
Comment (Vonsen) [inline LambdaEngine/Source/Resources/ResourceLoader.cpp]: When is this being used?

**Review by Vonsen — CHANGES_REQUESTED** [review]

**Review by Vonsen — APPROVED** [review]

**Pull request merged**